src/components/tests/assistant-code-block.test.tsx

```diff
@@ -0,0 +1,147 @@
+import { render, screen, fireEvent, act } from '@testing-library/react';
+import { getIpcApi } from '../../lib/get-ipc-api';
+import createCodeComponent from '../assistant-code-block';
+
+jest.mock( '../../lib/get-ipc-api' );
+
+describe( 'createCodeComponent', () => {
+	const contextProps = {
+		blocks: [],
+		updateMessage: jest.fn(),
+		projectPath: '/path/to/project',
+		messageId: 1,
+	};
+	const CodeBlock = createCodeComponent( contextProps );
+
+	it( 'should render inline styles for language-generic code', () => {
+		render( <CodeBlock children="example-code" /> );
+
+		expect( screen.getByText( 'example-code' ) ).toBeInTheDocument();
+		expect( screen.queryByText( 'Copy' ) ).not.toBeInTheDocument();
+		expect( screen.queryByText( 'Run' ) ).not.toBeInTheDocument();
+	} );
+
+	it( 'should display a "copy" button for language-specific code', () => {
+		render( <CodeBlock className="language-bash" children="wp --version" /> );
+
+		expect( screen.getByText( 'Copy' ) ).toBeInTheDocument();
+	} );
+
+	it( 'should display the "run" button for elligble wp-cli commands without placeholder content', () => {
+		render( <CodeBlock className="language-bash" children="wp --version" /> );
+
+		expect( screen.getByText( 'Run' ) ).toBeInTheDocument();
+	} );
+
+	it( 'should hide the "run" button for inellible non-wp-cli code', () => {
```

Comment:
Not sure if there's a typo in the word `inellible`, should be `ineligible`?

Reply:
Thanks for catching this typo. Addressed in https://github.com/Automattic/studio/pull/305/commits/2ae36fd29f28d6ce25b93887c7584a476521f2b7.